File: java/bubbleSort.java
import java.util.*;
import java.io.*;

public class bubbleSort{
	public static void main(String args[]){
		int[] unsorted = {6,1,2,5,3,8,4,7};
		int[] sorted = bubbleSort(unsorted);
		System.out.println(Arrays.toString(sorted));
	}

	public static int[] bubbleSort(int[] arr){
		boolean sorted = false;
		int temp;
		while(sorted == false){
			sorted = true;
			for(int i=0; i<arr.length-1; i++){
				if(arr[i] > arr[i+1]){
					temp = arr[i];
					arr[i] = arr[i+1];
					arr[i+1] = temp;
					sorted = false;
				}
			}
		}
		return arr;
	}
}

File: python/sorting/bubble_sort.py
def bubble_sort(arr):
	for i in range(len(arr)):
		print i
		done = True
		for i in range(len(arr))[:len(arr)-1]:
			if arr[i] > arr[i+1]:
				temp = arr[i]
				arr[i] = arr[i+1]
				arr[i+1] = temp
				done = False
			print arr
		print done
		if done:
			return 

arr = [5,4,3,2,1]
bubble_sort(arr)
print arr
